quantum/backlight/backlight_chibios.c

```diff
@@ -40,16 +40,24 @@
 #    endif
 #endif
 
-static PWMConfig pwmCFG = {0xFFFF, /* PWM clock frequency  */
-                           256,    /* PWM period (in ticks) 1S (1/10kHz=0.1mS 0.1ms*10000 ticks=1S) */
-                           NULL,   /* Breathing Callback */
-                           {       /* Default all channels to disabled - Channels will be configured durring init */
-                            {PWM_OUTPUT_DISABLED, NULL},
-                            {PWM_OUTPUT_DISABLED, NULL},
-                            {PWM_OUTPUT_DISABLED, NULL},
-                            {PWM_OUTPUT_DISABLED, NULL}},
-                           0, /* HW dependent part.*/
-                           0};
+#ifndef BACKLIGHT_PWM_COUNTER_FREQUENCY
+#    define BACKLIGHT_PWM_COUNTER_FREQUENCY 0xFFFF
+#endif
+
+#ifndef BACKLIGHT_PWM_PERIOD
+#    define BACKLIGHT_PWM_PERIOD 256
+#endif
```

Comment:
Making this tunable is a great feature by itself (256 Hz PWM lighting looks like crap, even the shittiest old WS2812B use ≈400 Hz, and that looks bad too). The only concern is that exposing the tunable parameters this way may not be the most understandable one, but restrictions of the actual hardware are mostly on these values, so maybe such tunables would be acceptable. (The AVR driver currently has only the `BACKLIGHT_CUSTOM_RESOLUTION` tunable that controls the PWM output frequency indirectly.)

Reply:
Nice! Previously the advance of backlight steps was done in the PWM channel wrap callback, which of course would either slow down or speed up depending on the chosen PWM period. This is now done using a virtual timer which runs at ~256Hz so in line with the old cone and decouples the advance from the PWM settings.